sormas-backend/src/main/java/de/symeda/sormas/backend/sample/PathogenTestFacadeEjb.java

```diff
@@ -68,274 +72,316 @@
+        target.setTestTypeText(source.getTestTypeText());
+        target.setTestDateTime(source.getTestDateTime());
+        target.setLab(facilityService.getByReferenceDto(source.getLab()));
+        target.setLabDetails(source.getLabDetails());
+        target.setLabUser(userService.getByReferenceDto(source.getLabUser()));
+        target.setTestResult(source.getTestResult());
+        target.setTestResultText(source.getTestResultText());
+        target.setTestResultVerified(source.getTestResultVerified());
+        target.setFourFoldIncreaseAntibodyTiter(source.isFourFoldIncreaseAntibodyTiter());
+        target.setSerotype(source.getSerotype());
+        target.setCqValue(source.getCqValue());
+
+        return target;
+    }
+
+    public PathogenTestDto toDto(PathogenTest source) {
+        if (source == null) {
+            return null;
+        }
+        PathogenTestDto target = new PathogenTestDto();
+        DtoHelper.fillDto(target, source);
+
+        target.setSample(SampleFacadeEjb.toReferenceDto(source.getSample()));
+        target.setTestedDisease(source.getTestedDisease());
+        target.setTestedDiseaseDetails(source.getTestedDiseaseDetails());
+        target.setTestType(source.getTestType());
+        target.setTestTypeText(source.getTestTypeText());
+        target.setTestDateTime(source.getTestDateTime());
+        target.setLab(FacilityFacadeEjb.toReferenceDto(source.getLab()));
+        target.setLabDetails(source.getLabDetails());
+        target.setLabUser(UserFacadeEjb.toReferenceDto(source.getLabUser()));
+        target.setTestResult(source.getTestResult());
+        target.setTestResultText(source.getTestResultText());
+        target.setTestResultVerified(source.getTestResultVerified());
+        target.setFourFoldIncreaseAntibodyTiter(source.isFourFoldIncreaseAntibodyTiter());
+        target.setSerotype(source.getSerotype());
+        target.setCqValue(source.getCqValue());
+
+        return target;
+    }
+
+    private void onPathogenTestChanged(PathogenTestDto existingPathogenTest, PathogenTest newPathogenTest) {
+        // Send an email to all responsible supervisors when a new non-pending sample test is created or the status of
+        // a formerly pending test result has changed
+        final String sampleUuid = newPathogenTest.getSample().getUuid();
+        final Sample sample = sampleService.getByUuid(sampleUuid);
+        final Case caze = sample.getAssociatedCase();
+        final Contact contact = sample.getAssociatedContact();
+        final Region region;
+        final Disease disease;
+        final List<User> messageRecipients;
+
+        if (caze != null) {
+            region = caze.getRegion();
+            disease = caze.getDisease();
+            messageRecipients = userService.getAllByRegionAndUserRoles(region,
+                    UserRole.SURVEILLANCE_SUPERVISOR, UserRole.CASE_SUPERVISOR);
+        } else {
+            region = contact.getRegion();
```

Comment:
Might be null, you have to use the source case region and district as a fallback